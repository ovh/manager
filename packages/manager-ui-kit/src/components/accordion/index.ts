export { Accordion } from './Accordion.component';
export type { AccordionProps } from './Accordion.props';

export { AccordionContent } from './accordion-content/AccordionContent.component';
export { AccordionItem } from './accordion-item/AccordionItem.component';
export { AccordionTrigger } from './accordion-trigger/AccordionTrigger.component';
