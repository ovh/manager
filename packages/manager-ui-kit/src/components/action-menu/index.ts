export { ActionMenu } from './ActionMenu.component';
export type { ActionMenuItem, ActionMenuProps } from './ActionMenu.props';
