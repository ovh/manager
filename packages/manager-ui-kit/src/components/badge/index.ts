export { Badge } from './badge.component';

export type { BadgeProps } from './badge.props';

export * from './Badge.constants';
