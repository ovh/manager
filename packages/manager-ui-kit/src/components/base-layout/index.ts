export { BaseLayout } from './BaseLayout.component';

export type { BaseLayoutProps } from './BaseLayout.props';

export type { HeaderProps } from './header/Header.props';
