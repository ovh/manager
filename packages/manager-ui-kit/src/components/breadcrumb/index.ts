export { Breadcrumb } from './Breadcrumb.component';
export type { BreadcrumbProps } from './Breadcrumb.props';
export * from './Breadcrumb.constants';
