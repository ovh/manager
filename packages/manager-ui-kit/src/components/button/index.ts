export { Button } from './Button.component';
export type { ButtonProps } from './Button.props';
