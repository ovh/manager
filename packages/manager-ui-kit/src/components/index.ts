export * from './action-banner';
export * from './action-menu';
export * from './redirection-guard';
export * from './breadcrumb';
export * from './base-layout';
export * from './clipboard';
export * from './step';
export * from './tabs';
export * from './tiles-input-group';
export * from './tiles-input';

export * from './Link';
export * from './drawer';
export * from './guide-menu';

export * from './notifications';

export * from './price';

export * from './filters';

export * from './button';
export * from './text';

export * from './service-state-badge';

export * from './tile';
export * from './order';

export * from './badge';
export * from './modal';
export * from './tags-list';
export * from './tags-tile';
export * from './changelog-menu';
export * from './link-card';
export * from './error';
export * from './error-boundary';
export * from './update-name-modal';
export * from './delete-modal';
export * from './grid-layout';
export * from './onboarding-layout';
export * from './datagrid';
export * from './accordion';
export * from './checkbox';
export * from './combobox';
export * from './datepicker';
export * from './file-upload';
export * from './form-field';
export * from './message';
export * from './phone-number';
export * from './popover';
export * from './progress';
export * from './quantity';
export * from './radio-group';
export * from './select';
export * from './switch';
export * from './timepicker';
export * from './toggle';
export * from './tooltip';
export * from './tree-view';
