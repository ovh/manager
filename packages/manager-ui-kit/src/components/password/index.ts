export { Password } from './Password.component';
export type { PasswordProps } from './Password.props';
