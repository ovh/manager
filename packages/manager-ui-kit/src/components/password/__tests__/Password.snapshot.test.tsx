import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Password } from '@/components';

describe('Password Snapshot tests', () => {
  it('renders the component with default props', () => {
    const { container } = render(
      <>
        <Password />
        <Password disabled />
        <Password defaultValue="Readonly" readOnly />
        <Password clearable defaultValue="Clearable" />
      </>,
    );
    expect(container).toMatchSnapshot();
  });
});
