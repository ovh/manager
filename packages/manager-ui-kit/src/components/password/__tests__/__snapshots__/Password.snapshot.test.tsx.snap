// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Password Snapshot tests > renders the component with default props 1`] = `
<div>
  <div
    class="_input_ff5b7_2 _input--with-actions_ff5b7_38"
    role="group"
  >
    <input
      aria-busy="false"
      class="_input__field_ff5b7_41"
      data-ods="password"
      id=":r0:"
      type="password"
      value=""
    />
    <div
      class="_input-actions_ke1hx_2"
    >
      <button
        aria-controls=":r0:"
        aria-label="Show password"
        class="_button_1koyw_2 _button--primary_1koyw_100 _button--xs_1koyw_25 _button--ghost_1koyw_151"
        data-ods="button"
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--eye-off_10c23_238"
          data-ods="icon"
          role="presentation"
        />
      </button>
    </div>
  </div>
  <div
    class="_input_ff5b7_2 _input--with-actions_ff5b7_38"
    data-disabled="true"
    role="group"
  >
    <input
      aria-busy="false"
      class="_input__field_ff5b7_41"
      data-ods="password"
      disabled=""
      id=":r1:"
      type="password"
      value=""
    />
    <div
      class="_input-actions_ke1hx_2"
    >
      <button
        aria-controls=":r1:"
        aria-label="Show password"
        class="_button_1koyw_2 _button--primary_1koyw_100 _button--xs_1koyw_25 _button--ghost_1koyw_151"
        data-ods="button"
        disabled=""
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--eye-off_10c23_238"
          data-ods="icon"
          role="presentation"
        />
      </button>
    </div>
  </div>
  <div
    class="_input_ff5b7_2 _input--with-actions_ff5b7_38"
    data-readonly="true"
    role="group"
  >
    <input
      aria-busy="false"
      class="_input__field_ff5b7_41"
      data-ods="password"
      id=":r2:"
      readonly=""
      type="password"
      value="Readonly"
    />
    <div
      class="_input-actions_ke1hx_2"
    >
      <button
        aria-controls=":r2:"
        aria-label="Show password"
        class="_button_1koyw_2 _button--primary_1koyw_100 _button--xs_1koyw_25 _button--ghost_1koyw_151"
        data-ods="button"
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--eye-off_10c23_238"
          data-ods="icon"
          role="presentation"
        />
      </button>
    </div>
  </div>
  <div
    class="_input_ff5b7_2 _input--with-actions_ff5b7_38"
    role="group"
  >
    <input
      aria-busy="false"
      class="_input__field_ff5b7_41"
      data-ods="password"
      id=":r3:"
      type="password"
      value="Clearable"
    />
    <div
      class="_input-actions_ke1hx_2"
    >
      <button
        aria-controls=":r3:"
        aria-label="Clear input"
        class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151"
        data-ods="button"
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--xmark_10c23_562"
          data-ods="icon"
          role="presentation"
        />
      </button>
      <button
        aria-controls=":r3:"
        aria-label="Show password"
        class="_button_1koyw_2 _button--primary_1koyw_100 _button--xs_1koyw_25 _button--ghost_1koyw_151"
        data-ods="button"
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--eye-off_10c23_238"
          data-ods="icon"
          role="presentation"
        />
      </button>
    </div>
  </div>
</div>
`;
